### PR TITLE
[DNM] Noir greyscale colours are now properly set and unset

### LINKED ISSUE #12080: Noir effect remains after clean SE and ryetalyn
(WEB REPORT BY: hudenagyon REMOTE: 172.93.109.202:7777)

> Revision (Should be above if you're viewing this from ingame!)
> 
> General description of the issue
> Roundstart noir effects are not affected by clean SE and ryetalyn even of the player gets the power deactivation message. Blood doesn't show up as red or blue or whatever color it has.
> 
> What you expected to happen
> Blood shouldn't be affected by monochrome affect. Deactivation of noir should remove the black and white.
> 
> What actually happened
> Read above. Additionally, restarting byond didn't help. Noir music started up and everything went gradually monochrome, including blood.
> 
> Steps to reproduce if possible
> roundstart gumshoe
> get clean se and ryetalin
> spread some blood around
> try to restart

### LINKED ISSUE #18491: Noir Vision (genetic defect) Bug
(WEB REPORT BY: deussex REMOTE: 172.93.109.202:7777)
# Revision
7da454228522c37dda9edbd8b65e48e0686562cb

# Description
I died, got cloned, got the black-and-white Noir vision as a genetic defect; complete with the sax music. But when I was put in cyro, my vision didn't revert to normal. I tried a ryetalin and that didn't work. When I got injected with a clean SE injection form genetics, that didn't work either. Now I'm stuck with black and white vision.
# Steps to Reproduce
1.)Die
2.) Get Cloned
   2A.) Hope you get the Noir vision upon being cloned
3.) Attempt to be cured via clean SE injection or Ryetalin
# What you Expected
My vision goes back to normal my means of genetic defect-curing. 
# What Actually Happened
I'm stuck with black and white vision, despite those attempts.

### LINKED ISSUE #19280: Mushroom man Invert Vision spell isn't reversible anymore
Inverted vision now stays forever until you relog, even if you ghost. Using the spell a second time does nothing now.

### LINKED ISSUE #19613: Softboiled color stays black and white
(WEB REPORT BY: vonzippenstein REMOTE: 172.93.109.202:7777)
# Revision
34f72c28e6be4f001e3c6b8bd05d1747097c7671

# Description
Vision went black and white after a genetics SE injection with the hardboiled noire detective gene. Reverted it with a clean block and my colors are still black and white.

# Steps to Reproduce
Inject yourself with the noire gene. Then inject yourself with a clean SE.

# What you Expected
Colors to return to normal and the message "You now feel softboiled."

# What Actually Happened
Just the message "You now feel softboiled" but you vision stays black and white.

### PR DESCRIPTION
**Pull request by BarneyGumball**
MERGE WHEN WE ARE ON `512.1454`.

Fixes #19613
Fixes #19280
Fixes #18491
Fixes #12080

* Removed unused `bad_changing_colour_ckeys` global var.  No idea what this was meant to do, but it was doing nothing.
* Removed the `updating_colour` var from client.  There's no need to track whether the client is going through a colour change animation or not because all the colour transition stuff is nicely handled by the `animate()` proc.
* Moved the noir specific stuff from the `update_colour()` proc to the appropriate noir file.
* Oh and the colour transitions have been fixed.

:cl:
 * bugfix: Losing colour visions (noir, shroom invert, etc) will now properly set colour overlays back to normal.  Update your client to the latest beta if you get weird colours after your vision resets.

### CODE REVIEW

File: code/modules/mob/mob_helpers.dm [hunk 1]
```diff
@@ -78,36 +78,18 @@ mob/proc/remove_internal_organ()
...
-	if(client.color)
-		difference = difflist(client.color,colour_to_apply)
-	if(!difference) // otherwise !difference.len throws a runtime since null.len isn't a thing
-		return
-	else if(!difference.len)
-		client.updating_colour = 1
-		var/cached_ckey = client.ckey
-		if(forceupdate)
-			time = 0
-		else if(colour_to_apply == NOIRMATRIX)
```
Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [inline code/modules/mob/mob_helpers.dm]: nice snowflake celt

**Comment by BarneyGumball** [comment]
This is ready to be merged once we are on `512.1454` and the secret repo conflicts are resolved.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
We're in 512.1454 aren't we

**Comment by BarneyGumball** [comment]
Not yet unfortunately:

**Pull request merged**